### DISCUSSION TITLE
Struggling with dynamic Texts and SVG Export

### DISCUSSION THREAD

**Discussion by psusset** [fabricjs/fabric.js#10019] [Q&A]
Hi All !
I'm having a lot of fun working with FabricJS but I'm facing an issue with the final step of my work.
I'm using FabricJs to let my customers create some certificate templates which will be rendered as PDF when their users will finish some activities.
I've created the Certificate Editor which works well and allow the creator to insert some variables like {{ username }}.
Once the canvas is saved, I generate a SVG from it and save it on database.
When I need to generate a certificate, I render the SVG in an html page and create a PDF of it on server side (this can't be on front side as certificates can be generated from a CRON job).
During the rendering step, I replace the variables with user specific data and it works well until the text is not left aligned.
My problem is that the replaced text will start at the exact position of the placeholder and after analysing the _ToSVG method of the textbox, the result I get is logic as the X position of each tspan is calculated from the characters included on it.
**I would like to get your thought about which strategy I can imagine to handle this issue** 

Thanks for your help on this,

Pierre

**Comment by psusset** [comment]
I finally changed my strategy and decided not to use SVG to generate the PDF.
As my canvas are quite simple (just images and texts), I rebuild the layout in html with divs (position:absolute) and I use the JSON to get all parameters for each objects. This gives me good results !

**Comment by asturur** [comment]
hi @psusset sorry for not answering before, but without a sample text and what kind of coordinates you were looking for i m not sure how to help.
I have some fabric to pdfkit code, but to position text absolutely perfect i need to convert text to paths

**Reply by psusset** [reply]
Hello @asturur, thanks for your reply. As I say in my second post, I finally get perfect result by reconstructing the layout with html and I was able to support a lot of text layout options (font, size, style, weight, color ...) even on part of the text. I create a JSON parser that creates html div and span according to text styles and the advantage of HTML compared to SVG is that when I replace the variables by user content, the result is still perfect.
